pkg/controller/backup_sync_controller.go

```diff
@@ -171,6 +172,27 @@ func (b *backupSyncReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 		backup.Labels[velerov1api.StorageLocationLabel] = label.GetValidName(backup.Spec.StorageLocation)
 
+		//check for the backup schedule. If it does not exist, remove it.
+		listedReferences := backup.ObjectMeta.OwnerReferences
+		foundReferences := make([]metav1.OwnerReference, 0)
+		for _, v := range listedReferences {
+			schedule := new(velerov1api.Schedule)
```

Comment:
should I be checking That the uid of the fetched and stored schedules are the same? it appears the other issue specifically references uid as an issue.

Reply:
Yes, check the UID either to make sure it's the same schedule